src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionTestCaseEventManagerFactory.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.CommunicationUtilities
+{
+    using System;
+
+    using Microsoft.VisualStudio.TestPlatform.CommunicationUtilities.DataCollection;
+    using Microsoft.VisualStudio.TestPlatform.CommunicationUtilities.Interfaces;
+
+    /// <summary>
+    /// Factory for creating sender and handler for interaction with datacollector process and test exectuion process.
+    /// </summary>
+    internal class DataCollectionTestCaseEventManagerFactory : IDataCollectionTestCaseEventManagerFactory
```

Comment:
Why are we using Factory? Please explain how this pattern fits here.
Also Sender is not implemented, hoping it will be taken up in the following commit.